src/baseSwagger.ts

```diff
@@ -287,6 +273,20 @@ const getPermissionDescription = (permissions: { [groupName: string]: string[] }
 	).join('')}</ul>`
 }
 
+const checkUniqueSharedSchema = (existingComponents: ComponentsSchema, newComponents: ComponentsSchema) => {
+	forEach(newComponents, (schemas, schemaType) => {
```

Comment:
@lubomir-dlhy should not be first iterate through existingComponents ? because if in schema does not exist any shared component (existingComponents ) still this funcion have to iterate through "newComponents"

Reply:
This function is called when swagger is generated from each Joi response schema / Joi request schema. It is expected that on average each request/response schema will have less new components than there are existing components, therefore in my opinion it is more efficient to iterate over new components. The newComponents is an empty object, if there is no .meta({className: '...'}) in the specific joi response/request schema. The existingComponents will gradually increase in size as more joi schemas are processed and new shared schemas are added.